File: Test-3 (27 Nov 2020)/Test/Q3. Counting CLosed Paths/CountingCLosedPaths.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strconv"
	"strings"
)




/*
 * Complete the 'closedPaths' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER number as parameter.
 */

func closedPaths(number int32) int32 {
	// Write your code here

	int numOfClosedPaths = 0
	int rem
	int num;

	num = number

	for num > 0 {
		rem = num % 10

		switch rem {
			case 0:
			case 4:
			case 6:
			case 9:
				numOfClosedPaths = numOfClosedPaths + 1
				break

			case 8:
				numOfClosedPaths = numOfClosedPaths + 2
				break			
		}

		num = num / 10
	}

	return numOfClosedPaths	
}





func main() {
	reader := bufio.NewReaderSize(os.Stdin, 16*1024*1024)

	stdout, err := os.Create(os.Getenv("OUTPUT_PATH"))
	checkError(err)

	defer stdout.Close()

	writer := bufio.NewWriterSize(stdout, 16*1024*1024)

	numberTemp, err := strconv.ParseInt(strings.TrimSpace(readLine(reader)), 10, 64)
	checkError(err)
	number := int32(numberTemp)

	result := closedPaths(number)

	fmt.Fprintf(writer, "%d\n", result)

	writer.Flush()
}

func readLine(reader *bufio.Reader) string {
	str, _, err := reader.ReadLine()
	if err == io.EOF {
		return ""
	}

	return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}

File: Test-3 (27 Nov 2020)/Test/Q3. Counting CLosed Paths/CountingCLosedPaths.py
#!/bin/python3

import math
import os
import random
import re
import sys



#
# Complete the 'closedPaths' function below.
#
# The function is expected to return an INTEGER.
# The function accepts INTEGER number as parameter.
#

def closedPaths(number):
    # Write your code here
   
   num2 = str(number)

    return (num2.count("0") + num2.count("4") + num2.count("6") + num2.count("9") + (2 * num2.count("8")))

"""
alternate method:

num = number
numOfClosedPaths = 0
rem = 0

for i in range (0, len(str(num))):

    rem = num % 10

    if rem == 0 or rem == 4 or rem == 6 or rem == 9:
        numOfClosedPaths = numOfClosedPaths + 1
    elif rem == 8:
        numOfClosedPaths = numOfClosedPaths + 2

    num = int(num / 10)

    i = i + 1
"""


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    number = int(input().strip())

    result = closedPaths(number)

    fptr.write(str(result) + '\n')

    fptr.close()



